File: CupcakeShop.Client/src/store/types/order.ts
import { ShortCartItem } from "./shortCartItem";

export type Order = {
    name?: string;
    phone?: string;
    email?: string;
    date?: string;
    deliveryMethod?: string;
    city?: string;
    street?: string;
    house?: string;
    entrance?: string;
    building?: string;
    apartment?: string;
    floor?: string;
    paymentMethod?: string;
    commentary?: string;
    cart?: ShortCartItem[];
}

File: CupcakeShop.Core/Entities/Order.cs
namespace CupcakeShop.Core.Entities
{
    public class Order : BaseEntity
    {
        public Order()
        {
            OrderedProducts = new List<OrderedProduct>();
        }
        public string Name { get; set; } = string.Empty;
        public string Phone { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Date { get; set; } = string.Empty;
        public string DeliveryMethod { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string Street { get; set; } = string.Empty;
        public string House { get; set; } = string.Empty;
        public string Entrance { get; set; } = string.Empty;
        public string Building { get; set; } = string.Empty;
        public string Apartment { get; set; } = string.Empty;
        public string Floor { get; set; } = string.Empty;
        public string PaymentMethod { get; set; } = string.Empty;
        public string Commentary { get; set; } = string.Empty;

        public List<OrderedProduct> OrderedProducts { get; set; }
    }

}
